NewApp/ngsa-csharp/Ngsa.DataService.Tests/AppTests.cs

```diff
@@ -96,10 +95,22 @@ public async Task RunApp()
                     d.Dispose();
                 }
 
-                // end the app
-                t.Wait(App.InMemory ? 5000 : 35000);
+                Stopwatch sw = new Stopwatch();
+                sw.Start();
+
+                // wait up to 45 seconds for the file semaphore
+                while (sw.ElapsedMilliseconds < 45000)
+                {
+                    if (File.Exists("../../../../tests-complete"))
+                    {
+                        break;
+                    }
 
-                Console.WriteLine("Web server stopped");
+                    await Task.Delay(1000);
+                }
+
+                // end the app
+                t.Wait(1);
             }
```

Comment:
This same loop exists twice.  Should it move to a common function in a static class?

Reply:
We could. The code is in two different apps, so we would have to create a "shared test code assembly". It seems like additional complexity for minimal gain unless there is other code we can share.

My suggestion is to leave as-is and add a task to the board to look at how much shared code there is. If there's enough, create the shared library.

@aflinchb  - any thoughts?

Reply:
I agree with keeping it separate for now unless there are other opportunities to break out more shared test code - not aware of a simpler solution either.